### PR TITLE
[fuchsia] Fix tests; remove fuchsia dart SDK deps

### PR DESCRIPTION
**Pull request by arbreng**
Dart is being removed from the Fuchsia SDK, so remove references to it here.

Also remove references to workstation_eng and replace with terminal. workstation_eng is no longer a thing.

### CODE REVIEW

File: shell/platform/fuchsia/dart_runner/tests/fidl/echo.fidl [hunk 1]
```diff
@@ -1,8 +1,8 @@
-// Copyright 2019 The Fuchsia Authors. All rights reserved.
+// Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (miguelfrde) [inline shell/platform/fuchsia/dart_runner/tests/fidl/echo.fidl:1]: 2023? Should this change be undone?

File: shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_echo_server/BUILD.gn [hunk 2]
```diff
@@ -16,13 +16,7 @@ dart_library("lib") {
...
   sources = [ "main.dart" ]
 
-  deps = [
-    "//flutter/shell/platform/fuchsia/dart_runner/tests/fidl/flutter.example.echo:echo",
-    "//flutter/tools/fuchsia/dart:fidl",
-    "//flutter/tools/fuchsia/dart:fuchsia_services",
-    "//flutter/tools/fuchsia/fidl:fuchsia.logger",
-    "//flutter/tools/fuchsia/fidl:fuchsia.test",
-  ]
+  deps = []
```
Comment (miguelfrde) [inline shell/platform/fuchsia/dart_runner/tests/startup_integration_test/dart_echo_server/BUILD.gn:19]: nit: I think you can omit `deps` when it's empty.

File: shell/platform/fuchsia/dart_runner/tests/fidl/echo.fidl (hunk 1 above)
Comment (arbreng) [reply shell/platform/fuchsia/dart_runner/tests/fidl/echo.fidl:1]: All of the files in this repo use the flutter license, which for some reason starts with "Copyright 2013" :shrug: 

I just did this to make the license checker happy

File: shell/platform/fuchsia/dart_runner/dart_component_controller.cc [hunk 3]
```diff
@@ -213,6 +210,78 @@ bool DartComponentController::CreateAndBindNamespace() {
...
+                           << "): " << zx_status_get_string(status);
+          }
+        }
+      };
+  dart_outgoing_dir_ptr_to_check_on_open_.set_error_handler(
+      [this](zx_status_t status) {
+        dart_outgoing_dir_ptr_to_check_on_open_.Unbind();
+      });
+
+  composed_service_dir->AddService(
```
[the remark this answers is not in the document]
Comment (arbreng) [reply shell/platform/fuchsia/dart_runner/dart_component_controller.cc:268]: Discussed offline, and a comment explaining this is sufficient.  Added one

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**